account_bank_statement_import_online/tests/test_account_bank_statement_import_online.py

```diff
@@ -267,6 +267,48 @@ def test_interval_type_hours(self):
         provider.interval_type = 'hours'
         provider._compute_update_schedule()
 
+    def test_schedule_adjustment(self):
+        journal = self.AccountJournal.create({
+            'name': 'Bank',
+            'type': 'bank',
+            'code': 'BANK',
+            'bank_statements_source': 'online',
+            'online_bank_statement_provider': 'dummy',
+        })
+        provider = journal.online_bank_statement_provider_id
+        provider.active = True
+        provider.interval_type = 'hours'
+        provider.adjust_next_run = True
+        five_minutes_ago = self.now - relativedelta(minutes=5)
+        three_hours_earlier = five_minutes_ago - relativedelta(hours=3)
+        fiftyfive_minutes_later = five_minutes_ago + relativedelta(hours=1)
+        provider.next_run = three_hours_earlier
+        provider.last_successful_run = three_hours_earlier - relativedelta(hours=1)
+        provider._scheduled_pull()
+        self.assertEqual(provider.last_successful_run, five_minutes_ago)
+        self.assertEqual(provider.next_run, fiftyfive_minutes_later)
```

Comment:
To be on the super-safe size, please also check that number of pulled statement lines matches the expectation in both cases, whatever that number is

Reply:
Little bit complicated as I check for hours of delay but statements are created daily, and the number of statements created might depend on the time the tests run (so either 1 or 2 actual dates might be involved).